### PR TITLE
MDEXP-257 Log errors for inventory Instance record exports

### PR DESCRIPTION
**Pull request by illia-borysenko**
## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/service/export/storage/AWSStorageServiceImpl.java [hunk 1]
```diff
@@ -106,10 +113,20 @@ public void storeFile(FileDefinition fileDefinition, String tenantId) {
...
         LOGGER.error(e.getMessage());
+        populateErrorLog("S3 connection issue", fileDefinition.getJobExecutionId(), tenantId);
         Thread.currentThread().interrupt();
       } finally {
         transferManager.shutdownNow();
       }
     }
   }
+
+  private void populateErrorLog(String reason, String jobExecutionId, String tenantId) {
```
Comment (andrii-novytskyi) [inline src/main/java/org/folio/service/export/storage/AWSStorageServiceImpl.java:124]: looks like this method will be used in all classes where we will populate error log, worth thinking to move it to ErrorLogService

**Review by andrii-novytskyi — APPROVED** [review]

File: src/main/java/org/folio/service/export/storage/AWSStorageServiceImpl.java [hunk 2]
```diff
@@ -106,10 +113,20 @@ public void storeFile(FileDefinition fileDefinition, String tenantId) {
         multipleFileUpload.waitForCompletion();
       } catch (InterruptedException e) {
         LOGGER.error(e.getMessage());
+        populateErrorLog("S3 connection issue", fileDefinition.getJobExecutionId(), tenantId);
```
Comment (Igor-Gorchakov) [inline src/main/java/org/folio/service/export/storage/AWSStorageServiceImpl.java:116]: This is a really rare case when InterruptedException is thrown, usually, it happens when the application gets shutdown. No need to populate errors here

**Review by Igor-Gorchakov — APPROVED** [review]

File: src/main/java/org/folio/service/logs/ErrorLogService.java [hunk 3]
```diff
@@ -41,4 +41,9 @@
    * @return future with {@link ErrorLog}
    */
   Future<Boolean> deleteById(String id, String tenantId);
+
+  Future<ErrorLog> saveGeneralError(String reason, String jobExecutionId, String tenantId);
```
Comment (Igor-Gorchakov) [inline src/main/java/org/folio/service/logs/ErrorLogService.java:45]: Javadoc

File: src/main/java/org/folio/service/logs/ErrorLogServiceImpl.java [hunk 4]
```diff
@@ -22,20 +30,53 @@
...
+      .withLogLevel(ErrorLog.LogLevel.ERROR)
+      .withJobExecutionId(jobExecutionId);
+    return save(errorLog, tenantId);
+  }
+
+  @Override
+  public Future<ErrorLog> saveWithAffectedRecord(JsonObject record, String reason, String jobExecutionId, String tenantId) {
+    AffectedRecord instanceRecord = new AffectedRecord()
+      .withRecordType(INSTANCE);
+    if(StringUtils.isNotBlank(record.getString(HRID_KEY))) {
```
Comment (Igor-Gorchakov) [inline src/main/java/org/folio/service/logs/ErrorLogServiceImpl.java:64]: please format java code

**Comment by KVupp** [comment]
Please add what was done in the PR description, it will help to understand the PR details

File: src/main/java/org/folio/service/logs/ErrorLogServiceImpl.java [hunk 5]
```diff
@@ -22,20 +30,53 @@
   }
 
   @Override
-  public Future<ErrorLog> save(ErrorLog errorLog, OkapiConnectionParams params) {
+  public Future<ErrorLog> save(ErrorLog errorLog, String tenantId) {
     if (errorLog.getId() == null) {
       errorLog.setId(UUID.randomUUID().toString());
     }
-    return errorLogDao.save(errorLog, params.getTenantId());
+    errorLog.setCreatedData(new Date());
```
Comment (ViktorSoroka07) [inline src/main/java/org/folio/service/logs/ErrorLogServiceImpl.java:37]: It should be really `setCreatedDate`. And also the field should be named `createDate`.

**Pull request merged**